src/python/pants/backend/docker/docker_build_context_test.py

```diff
@@ -1,79 +1,89 @@
 # Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
-from pants.backend.docker.docker_build_context import (
-    DockerBuildContextRequest,
-    create_docker_build_context,
-)
+from __future__ import annotations
+
+from textwrap import dedent
+
+import pytest
+
+from pants.backend.docker.docker_build_context import DockerBuildContext, DockerBuildContextRequest
+from pants.backend.docker.rules import rules
 from pants.backend.docker.target_types import DockerImage
-from pants.backend.python.goals.package_pex_binary import PexBinaryFieldSet
-from pants.backend.python.target_types import PexBinary
-from pants.core.goals.package import BuiltPackage, PackageFieldSet
-from pants.core.util_rules.source_files import SourceFiles, SourceFilesRequest
+from pants.core.target_types import Files
+from pants.core.util_rules.source_files import rules as source_files_rules
 from pants.engine.addresses import Address
-from pants.engine.fs import EMPTY_DIGEST, EMPTY_SNAPSHOT, AddPrefix, Digest, MergeDigests
-from pants.engine.target import (
-    FieldSetsPerTarget,
-    FieldSetsPerTargetRequest,
-    TransitiveTargets,
-    TransitiveTargetsRequest,
-)
-from pants.engine.unions import UnionMembership
-from pants.testutil.rule_runner import MockGet, run_rule_with_mocks
+from pants.engine.fs import Snapshot
+from pants.testutil.rule_runner import QueryRule, RuleRunner
 
 
-def test_create_docker_build_context():
```

Comment:
I dropped the whole mocked test, as it was no more than a smoke test that the rule didn't simply crash.